msal/src/main/java/com/microsoft/identity/nativeauth/statemachine/states/AccountState.kt

```diff
@@ -102,63 +105,75 @@ class AccountState private constructor(
+                requestAccountRecord.homeAccountId = account.homeAccountId
+
+                val params = CommandParametersAdapter.createRemoveAccountCommandParameters(
+                    config,
+                    config.oAuth2TokenCache,
+                    requestAccountRecord
+                )
 
-            val result = CommandDispatcher.submitSilentReturningFuture(removeCurrentAccountCommandParameters)
-                .get().result as Boolean
+                val removeCurrentAccountCommandParameters = RemoveCurrentAccountCommand(
+                    params,
+                    LocalMSALController().asControllerFactory(),
+                    object : CommandCallback<Boolean?, BaseException?> {
+                        override fun onError(error: BaseException?) {
+                            // Do nothing, handled by CommandDispatcher.submitSilentReturningFuture()
+                        }
+
+                        override fun onTaskCompleted(result: Boolean?) {
+                            // Do nothing, handled by CommandDispatcher.submitSilentReturningFuture()
+                        }
+
+                        override fun onCancel() {
+                            // Do nothing
+                        }
+                    },
+                    PublicApiId.NATIVE_AUTH_ACCOUNT_SIGN_OUT
+                )
 
-            return@withContext if (result) {
-                SignOutResult.Complete
-            } else {
-                Logger.error(
-                    TAG,
-                    "Unexpected error during signOut.",
-                    null
+                val result = CommandDispatcher.submitSilentReturningFuture(
+                    removeCurrentAccountCommandParameters
                 )
-                throw MsalClientException(
-                    MsalClientException.UNKNOWN_ERROR,
-                    "Unexpected error during signOut."
+                    .get().result as Boolean
+
+                return@withContext if (result) {
+                    SignOutResult.Complete
+                } else {
+                    Logger.error(
+                        TAG,
+                        "Unexpected error during signOut.",
+                        null
+                    )
+                    throw MsalClientException(
+                        MsalClientException.UNKNOWN_ERROR,
+                        "Unexpected error during signOut."
+                    )
+                }
+            } catch (e: Exception) {
+                Logger.error(TAG, "MSAL client exception occurred in getCurrentAccount.", e)
+                ClientExceptionError(
+                    errorType = ErrorTypes.CLIENT_EXCEPTION,
+                    errorMessage = "MSAL client exception occurred in getCurrentAccount.",
+                    exception = e,
+                    correlationId = DiagnosticContext.INSTANCE.threadCorrelationId
```

Comment:
Setting a correlation ID here is wrong. It should already have been set earlier in the flow, and the existing value should be used.

Reply:
Use "UNSET" here right now.